libs/core/src/lib/button/base-button.ts

```diff
@@ -77,9 +77,27 @@ export class BaseButton {
 
     set ariaDisabled(value: boolean) { this._ariaDisabled = coerceBooleanProperty(value); }
 
+    /** Adding aria label to the component */
+    @Input()
+    set ariaLabel(ariaLabel: string) {
+        this._ariaLabel = ariaLabel;
+    }
+    get ariaLabel(): string {
+        return this._ariaLabel;
+    }
+
+    /** @hidden */
+    private _ariaLabel: string;
+
     /** @hidden */
     _disabled: boolean;
 
     /** @hidden */
     _ariaDisabled: boolean;
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        const ariaLabelChange = this.ariaLabel === undefined ? this.label : this.ariaLabel;
+        this._ariaLabel = ariaLabelChange + ' ' + this.fdType;
```

Comment:
Appending the `this.fdType` works for English but will not translate to other languages.  We need to provide a way for the application developer to set the entire label and adequately communicate in the documentation that the responsibility to handle translations is on them because it is out of the hands of the library